src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4219,7 +4219,14 @@ LogicalResult ONNXScatterElementsOp::inferShapes(
 
 LogicalResult ONNXScatterNDOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  if (!data().getType().isa<mlir::RankedTensorType>() ||
+      !indices().getType().isa<mlir::RankedTensorType>() ||
```

Comment:
Since the output shape depends on `data()`'s shape only. It's redundant to check `indices()` and `updates()`

Reply:
ok